module/zfs/vdev_file.c

```diff
@@ -210,6 +210,15 @@ vdev_file_io_start(zio_t *zio)
     vdev_t *vd = zio->io_vd;
     vdev_file_t *vf = vd->vdev_tsd;
     ssize_t resid = 0;
+#ifdef DEBUG
+    uint64_t zio_size_in = zio->io_size;
+    uint64_t abd_size_in = zio->io_abd->abd_size;
+
+    if (zio_size_in != abd_size_in) {
```

Comment:
I think this can be demoted back to an ASSERT3S soon; I can't reproduce it on any of my pools, although the "minefield" pool that is doing a double resilver is still not quite thoroughly testable.